### PR TITLE
fix: restrict "as" prop to allow only HTML elements

### PR DESCRIPTION
**Pull request by layershifter**
#### Pull request checklist

- [ ] ~Addresses an existing issue: Fixes #0000~
- [x] Include a change request file using `$ yarn change`

#### Description of changes

A follow up for RFC #17464.

### CODE REVIEW

File: packages/react-utilities/src/compose/types.ts [hunk 1]
```diff
@@ -6,13 +6,8 @@ import * as React from 'react';
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export type GenericDictionary = Record<string, any>;
 
-/**
- * Class dictionary.
- */
-export type ClassDictionary = Record<string, string>;
```
Comment (layershifter) [inline packages/react-utilities/src/compose/types.ts:12]: This was not used anywhere.

**Review by bsunderhus — APPROVED** [review]
LGTM

File: packages/react-utilities/src/compose/types.ts [hunk 2]
```diff
@@ -76,6 +71,6 @@ export type ComponentState<
   ShorthandProps extends keyof Props = never,
   DefaultedProps extends keyof ResolvedShorthandProps<Props, ShorthandProps> = never
 > = RequiredProps<ResolvedShorthandProps<Props, ShorthandProps>, DefaultedProps> & {
-  as?: React.ElementType;
+  as?: keyof JSX.IntrinsicElements;
```
Comment (behowell) [inline packages/react-utilities/src/compose/types.ts:74]: I don't recall why `as` is duplicated here, but I don't think it should be necessary (and is not desirable), given that it exists on `ComponentProps`. Does anything break if you remove it here?

**Review by behowell — APPROVED** [review]

**Comment by layershifter** [comment]
#17434 also plans to restrict a set of tags, closing in a favor of it.

FYI @bsunderhus

**Pull request closed without merging**